bgpd/bgp_vty.c

```diff
@@ -12739,6 +12739,10 @@ static void bgp_ac_neighbor(vector comps, struct cmd_token *token)
 			enum cmd_token_type match_type;
 			char *name = peer->host;
 
+			if (bgp_nexthop_fqdn(peer)
+			    && strncmp(token->varname, "neighbors", 9) == 0)
+				name = peer->hostname;
```

Comment:
Do we need to also set the `match_type` to an appropriate value? It looks like this will resolve `match_type` to `IPV4_TKN` if I'm not mistaken.

Reply:
During my tests it works as expected. It resolves to IPV6_TKN either if `:` is found in `peer->host`. Or I missed something?
```
exit1# show ip bgp summary 
...
Neighbor                           V         AS MsgRcvd MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd
10.10.10.200                       4          1       0       0        0    0    0    never       Active
10.10.10.201                       4          2       0       0        0    0    0    never       Active
spine1-debian-9(192.168.0.2)       4        100    2092    2019        0    0    0 00:02:40            9
2a02:4780::1                       4        100       0       0        0    0    0    never       Active
spine1-debian-9(2a02:4780:face::1) 4        100    2103    2027        0    0    0 00:02:40            9

Total number of neighbors 5
exit1# show bgp neighbors 
  <cr>      
  A.B.C.D   Neighbor to display information about
     10.10.10.200 10.10.10.201 spine1-debian-9 
  WORD      Neighbor on BGP configured interface
  X:X::X:X  Neighbor to display information about
     2a02:4780::1 spine1-debian-9 
  json      JavaScript Object Notation
exit1# show bgp neighbors spine1-debian-9
BGP neighbor is 192.168.0.2, remote AS 100, local AS 200, external link
Hostname: spine1-debian-9
  BGP version 4, remote router ID 172.16.1.1, local router ID 172.16.20.254
  BGP state = Established, up for 00:03:46
  Last read 00:00:01, Last write 00:00:00
  Hold time is 3, keepalive interval is 1 seconds
  Configured hold time is 3, keepalive interval is 1 seconds
  Neighbor capabilities:
    4 Byte AS: advertised and received
    AddPath:
      IPv4 Unicast: RX advertised IPv4 Unicast and received
    Route refresh: advertised and received(old & new)
    Address Family IPv4 Unicast: advertised and received
    Hostname Capability: advertised (name: exit1-debian-pati-geriausia-os-for-networking,domain name: n/a) received (name: spine1-debian-9,domain name: n/a)
    Graceful Restart Capabilty: advertised and received
      Remote Restart timer is 120 seconds
      Address families by peer:
        none
  Graceful restart information:
    End-of-RIB send: IPv4 Unicast
    End-of-RIB received: IPv4 Unicast
  Message statistics:
    Inq depth is 0
    Outq depth is 0
                         Sent       Rcvd
    Opens:                 11         11
    Notifications:          0         20
    Updates:               55        110
    Keepalives:          2019       2016
    Route Refresh:          0          0
    Capability:             0          0
    Total:               2085       2157
  Minimum time between advertisement runs is 0 seconds

 For address family: IPv4 Unicast
  Update group 32, subgroup 32
  Packet Queue length 0
  NEXT_HOP is always this router
  Community attribute sent to this neighbor(all)
  9 accepted prefixes

  Connections established 11; dropped 10
  Last reset 00:03:47, due to NOTIFICATION received (Cease/Peer Unconfigured)
Local host: 192.168.0.1, Local port: 179
Foreign host: 192.168.0.2, Foreign port: 34902
Nexthop: 192.168.0.1
Nexthop global: 2a02:4780:face::2
Nexthop local: fe80::a00:27ff:fecf:a756
BGP connection: shared network
BGP Connect Retry Timer in Seconds: 120
Read thread: on  Write thread: on  FD used: 24

BGP neighbor is 2a02:4780:face::1, remote AS 100, local AS 200, external link
Hostname: spine1-debian-9
  BGP version 4, remote router ID 172.16.1.1, local router ID 172.16.20.254
  BGP state = Established, up for 00:03:46
  Last read 00:00:01, Last write 00:00:00
  Hold time is 3, keepalive interval is 1 seconds
  Configured hold time is 3, keepalive interval is 1 seconds
  Neighbor capabilities:
    4 Byte AS: advertised and received
    AddPath:
      IPv4 Unicast: RX advertised IPv4 Unicast and received
      IPv6 Unicast: RX advertised IPv6 Unicast and received
    Route refresh: advertised and received(old & new)
    Address Family IPv4 Unicast: advertised and received
    Address Family IPv6 Unicast: advertised and received
    Hostname Capability: advertised (name: exit1-debian-pati-geriausia-os-for-networking,domain name: n/a) received (name: spine1-debian-9,domain name: n/a)
    Graceful Restart Capabilty: advertised and received
      Remote Restart timer is 120 seconds
      Address families by peer:
        none
  Graceful restart information:
    End-of-RIB send: IPv4 Unicast, IPv6 Unicast
    End-of-RIB received: IPv4 Unicast, IPv6 Unicast
  Message statistics:
    Inq depth is 0
    Outq depth is 0
                         Sent       Rcvd
    Opens:                 12         12
    Notifications:          2         20
    Updates:               82        142
    Keepalives:          1997       1994
    Route Refresh:          0          0
    Capability:             0          0
    Total:               2093       2168
  Minimum time between advertisement runs is 0 seconds

 For address family: IPv4 Unicast
  Update group 33, subgroup 33
  Packet Queue length 0
  Community attribute sent to this neighbor(all)
  9 accepted prefixes

 For address family: IPv6 Unicast
  Update group 34, subgroup 34
  Packet Queue length 0
  Community attribute sent to this neighbor(all)
  1 accepted prefixes

  Connections established 12; dropped 11
  Last reset 00:03:47, due to NOTIFICATION received (Cease/Peer Unconfigured)
Local host: 2a02:4780:face::2, Local port: 179
Foreign host: 2a02:4780:face::1, Foreign port: 55018
Nexthop: 192.168.0.1
Nexthop global: 2a02:4780:face::2
Nexthop local: fe80::a00:27ff:fecf:a756
BGP connection: shared network
BGP Connect Retry Timer in Seconds: 120
Read thread: on  Write thread: on  FD used: 25
```